File: testdata/vulnerable.py
import sqlite3
import os
import subprocess

# SQL Injection - BAD
def get_user(user_id):
    conn = sqlite3.connect("db.sqlite")
    cursor = conn.cursor()
    cursor.execute(f"SELECT * FROM users WHERE id = {user_id}")
    return cursor.fetchone()

# Command Injection - BAD
def ping_host(host):
    os.system(f"ping -c 1 {host}")

# Subprocess with shell=True - BAD
def run_command(cmd):
    subprocess.run(cmd, shell=True)

# Hardcoded secret - BAD
API_KEY = "sk-proj-1234567890abcdef1234567890abcdef"
DATABASE_PASSWORD = "super_secret_password_123"

# Safe parameterized query - GOOD (should not trigger)
def get_user_safe(user_id):
    conn = sqlite3.connect("db.sqlite")
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = ?", (user_id,))
    return cursor.fetchone()

File: testdata/vulnerable.js
const { exec } = require("child_process");
const mysql = require("mysql");

// SQL Injection - BAD
function getUser(userId) {
  const query = `SELECT * FROM users WHERE id = ${userId}`;
  return db.query(query);
}

// XSS - BAD
function renderContent(userInput) {
  document.getElementById("output").innerHTML = userInput;
  document.write(userInput);
}

// Command Injection - BAD
function runCommand(userInput) {
  exec(`ls ${userInput}`);
}

// Prototype Pollution - BAD
function mergeConfig(req) {
  Object.assign(config, req.body);
}

// Hardcoded Secret - BAD
const API_KEY = "sk-1234567890abcdef1234567890abcdef";

// dangerouslySetInnerHTML - BAD
function Component({ data }) {
  return <div dangerouslySetInnerHTML={{ __html: data }} />;
}

// Safe parameterized query - GOOD (should not trigger)
function getUserSafe(userId) {
  return db.query("SELECT * FROM users WHERE id = $1", [userId]);
}
